src/System.Private.CoreLib/shared/System/Threading/SemaphoreSlim.cs

```diff
@@ -348,28 +336,24 @@ public bool Wait(int millisecondsTimeout, CancellationToken cancellationToken)
             try
             {
                 // Perf: first spin wait for the count to be positive.
-                //       This additional amount of spinwaiting in addition
-                //       to Monitor.Enter()’s spinwaiting has shown measurable perf gains in test scenarios.
-                if (m_currentCount == 0)
+                // This additional amount of spinwaiting in addition
+                // to Monitor.Enter()’s spinwaiting has shown measurable perf gains in test scenarios.
+                // Monitor.Enter followed by Monitor.Wait is much more expensive than waiting on an event as it involves another
+                // spin, contention, etc. The usual number of spin iterations that would otherwise be used here is increased to
+                // lessen that extra expense of doing a proper wait.
+                var spinner = new SpinWait();
+#if CORECLR
```

Comment:
Is this ifdef really required? Looks like a performance fix that did not get ported to CoreRT.

Reply:
yeah you are right. https://github.com/dotnet/coreclr/pull/15002/commits/0799da25ac120d52a9878d8663382ca5590c954e 

is  checking the ```m_currentCount == 0```  in if condition rather than in while condition has some effect on performance  ?

Reply:
> `m_currentCount == 0` in if condition rather than in while condition has some effect on performance 

That alone - no. However, there are other differences.

Just keep the CoreCLR impl here. It is the right one.